AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncListDecoder.swift

```diff
@@ -9,16 +9,37 @@ import Foundation
 import Amplify
 
 public struct AppSyncListDecoder: ModelListDecoder {
+
+    /// Return true when the incoming payload to decode can be detected as one of the following scenarios:
```

Comment:
explaining the flow a bit here

previous conversation: https://github.com/aws-amplify/amplify-ios/pull/990/files#r548609876